include/cuda/std/detail/libcxx/include/experimental/__p0009_bits/config.hpp

```diff
@@ -0,0 +1,302 @@
+#    define __MDSPAN_USE_INLINE_VARIABLES 1
+#  endif
+#endif
+
+#ifndef __MDSPAN_NEEDS_TRAIT_VARIABLE_TEMPLATE_BACKPORTS
+#  if (!(defined(__cpp_lib_type_trait_variable_templates) && __cpp_lib_type_trait_variable_templates >= 201510L) \
+          || !__MDSPAN_HAS_CXX_17)
+#    if !(defined(__MDSPAN_COMPILER_APPLECLANG) && __MDSPAN_HAS_CXX_17)
+#      define __MDSPAN_NEEDS_TRAIT_VARIABLE_TEMPLATE_BACKPORTS 1
+#    endif
+#  endif
+#endif
+
+#ifndef __MDSPAN_USE_VARIABLE_TEMPLATES
+#  if (defined(__cpp_variable_templates) && __cpp_variable_templates >= 201304 && __MDSPAN_HAS_CXX_17) \
+        || (!defined(__cpp_variable_templates) && __MDSPAN_HAS_CXX_17)
+#    define __MDSPAN_USE_VARIABLE_TEMPLATES 1
+#  endif
+#endif // __MDSPAN_USE_VARIABLE_TEMPLATES
+
+#ifndef __MDSPAN_USE_CONSTEXPR_14
+#  if (defined(__cpp_constexpr) && __cpp_constexpr >= 201304) \
+        || (!defined(__cpp_constexpr) && __MDSPAN_HAS_CXX_14) \
+        && (!(defined(__INTEL_COMPILER) && __INTEL_COMPILER <= 1700))
+#    define __MDSPAN_USE_CONSTEXPR_14 1
+#  endif
+#endif
+
+#ifndef __MDSPAN_USE_INTEGER_SEQUENCE
+#  if defined(__MDSPAN_COMPILER_MSVC)
+#    if (defined(__cpp_lib_integer_sequence) && __cpp_lib_integer_sequence >= 201304)
+#      define __MDSPAN_USE_INTEGER_SEQUENCE 1
+#    endif
+#  endif
+#endif
+#ifndef __MDSPAN_USE_INTEGER_SEQUENCE
+#  if (defined(__cpp_lib_integer_sequence) && __cpp_lib_integer_sequence >= 201304) \
+        || (!defined(__cpp_lib_integer_sequence) && __MDSPAN_HAS_CXX_14) \
+        /* as far as I can tell, libc++ seems to think this is a C++11 feature... */ \
+        || (defined(__GLIBCXX__) && __GLIBCXX__ > 20150422 && __GNUC__ < 5 && !defined(__INTEL_CXX11_MODE__))
+     // several compilers lie about integer_sequence working properly unless the C++14 standard is used
+#    define __MDSPAN_USE_INTEGER_SEQUENCE 1
+#  elif defined(__MDSPAN_COMPILER_APPLECLANG) && __MDSPAN_HAS_CXX_14
+     // appleclang seems to be missing the __cpp_lib_... macros, but doesn't seem to lie about C++14 making
+     // integer_sequence work
+#    define __MDSPAN_USE_INTEGER_SEQUENCE 1
+#  endif
+#endif
+
+#ifndef __MDSPAN_USE_RETURN_TYPE_DEDUCTION
+#  if (defined(__cpp_return_type_deduction) && __cpp_return_type_deduction >= 201304) \
+          || (!defined(__cpp_return_type_deduction) && __MDSPAN_HAS_CXX_14)
+#    define __MDSPAN_USE_RETURN_TYPE_DEDUCTION 1
+#  endif
+#endif
+
+#ifndef __MDSPAN_USE_CLASS_TEMPLATE_ARGUMENT_DEDUCTION
+// GCC 10's CTAD seems sufficiently broken to prevent its use.
+#  if (defined(__MDSPAN_COMPILER_CLANG) || !defined(__GNUC__) || __GNUC__ >= 11) \
+      && ((defined(__cpp_deduction_guides) && __cpp_deduction_guides >= 201703) \
+         || (!defined(__cpp_deduction_guides) && __MDSPAN_HAS_CXX_17))
+#    define __MDSPAN_USE_CLASS_TEMPLATE_ARGUMENT_DEDUCTION 1
+#  endif
+#endif
+
+#ifndef __MDSPAN_USE_ALIAS_TEMPLATE_ARGUMENT_DEDUCTION
+// GCC 10's CTAD seems sufficiently broken to prevent its use.
```

Comment:
@youyu3 We may want to excise some of the original author's snark for public releases.

Reply:
To clarify: It's not a huge deal here (the comment is not too snarky) but we could change it to something like "GCC 10 is known not to work with CTAD for this case."

Reply:
Done.